engines/registry/app/models/access_token.rb

```diff
@@ -20,7 +20,7 @@ def claim
       hash['iss']    = 'RMT' # "matching issuer in registry auth token config"
       hash['sub']    = @account
       hash['aud']    = @service
-      hash['exp']    = Time.now.getlocal.to_i + (5 * 60) # expires at
+      hash['exp']    = Time.now.getlocal.to_i + Settings[:registry].token_expiration # expires at
```

Comment:
As far as I can remember, the docker/podman client was not re-using access tokens for subsequent commands, so this might not have an effect here. @cwh42 looked deeper into this, can you share what was the client behavior regarding token expiration?

Reply:
I have checked the behavior on testing infra server and it works as expected:
- running search/pull within the time frame before expiring => returns search or it is able to pull a container
- running search/pull outside the time frame                           => requires credentials refresh

Updated to use hardcoded or settings from file too